src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/ExportAzureRMBlueprint.cs

```diff
@@ -48,38 +49,45 @@ public class ExportAzureRmBlueprint : BlueprintDefinitionCmdletBase
+
+            // Get serialized blueprint definition
             string serializedDefinition = BlueprintClient.GetBlueprintDefinitionJsonFromObject(Blueprint, Version);
 
-            var blueprintFolderPath = CreateFolderIfNotExist(OutputPath, Blueprint.Name);
-            var blueprintJsonFilePath = Path.Combine(blueprintFolderPath, $"{Blueprint.Name}.json");
+            var resolvedPath = ResolveUserPath(OutputPath);
+
+            var blueprintFolderPath = Path.Combine(resolvedPath, Blueprint.Name);
 
+            // if directory exists, let's ask if we can clean contents of it. If directory doesn't exist, we'll create one.
             this.ConfirmAction(
-                this.Force || !AzureSession.Instance.DataStore.FileExists(blueprintJsonFilePath),
-                string.Format(Resources.OverwriteExistingOutputFileProcessMessage,Blueprint.Name),
-                Resources.OverwriteExistingOutputFileContinueMessage,
-                blueprintJsonFilePath,
-                () => AzureSession.Instance.DataStore.WriteFile(blueprintJsonFilePath, serializedDefinition)
+                this.Force || !AzureSession.Instance.DataStore.DirectoryExists(blueprintFolderPath),
+                string.Format(Resources.DeleteBlueprintFolderContentsProcessString, Blueprint.Name),
+                Resources.DeleteBlueprintFolderContentsContinueMessage,
+                blueprintFolderPath,
+                () => CreateFolderIfNotExist(resolvedPath, Blueprint.Name)
             );
+            
+            var blueprintJsonFilePath = Path.Combine(blueprintFolderPath, $"{blueprintFileName}.json");
 
-            // Get serialized artifacts from this blueprint and write them to disk
-            var artifactsPath = CreateFolderIfNotExist(blueprintFolderPath, "Artifacts");
-
+            AzureSession.Instance.DataStore.WriteFile(blueprintJsonFilePath, serializedDefinition);
+           
             var artifacts = BlueprintClient.ListArtifacts(Blueprint.Scope, Blueprint.Name, Version);
-            
-            foreach (var artifact in artifacts)
+
+            if (artifacts != null && artifacts.Any())
             {
-                string serializedArtifact = BlueprintClient.GetBlueprintArtifactJsonFromObject(Blueprint.Scope, Blueprint.Name, artifact, Version);
+                // Get serialized artifacts from this blueprint and write them to disk
+                var artifactsPath = CreateFolderIfNotExist(blueprintFolderPath, "Artifacts");
+
+                foreach (var artifact in artifacts)
+                {
+                    string serializedArtifact =
+                        BlueprintClient.GetBlueprintArtifactJsonFromObject(Blueprint.Scope, Blueprint.Name, artifact,
+                            Version);
 
-                var artifactFilePath = Path.Combine(artifactsPath, artifact.Name + ".json");
+                    var artifactFilePath = Path.Combine(artifactsPath, artifact.Name + ".json");
 
-                this.ConfirmAction(
```

Comment:
@filizt do we not need this check `ConfirmAction` check anymore?

Reply:
We don't since the "Artifacts" folder would have been deleted in the first blueprint creation step, which is the first confirm action, should it already existed locally.